### PR TITLE
feat: support 'disabled' prop [SL-1725]

### PR DESCRIPTION
**Pull request by billiegoose**

### CODE REVIEW

File: src/components/Label.tsx [hunk 1]
```diff
@@ -9,10 +9,16 @@ interface ILabel extends IText<HTMLLabelElement> {
...
 
-export const Label: React.FunctionComponent<ILabel> = ({ htmlFor, variant, children }) => {
+export const Label: React.FunctionComponent<ILabel> = ({ htmlFor, variant, children, disabled = false }) => {
   const fg = useFg(variant);
+  const disabledStyles = disabled
+    ? {
+        opacity: 0.6,
+        cursor: 'not-allowed',
+      }
+    : {};
```
Comment (P0lip) [inline src/components/Label.tsx]:
```suggestion
    : null;
```

File: src/components/Label.tsx [hunk 2]
```diff
@@ -9,10 +9,16 @@ interface ILabel extends IText<HTMLLabelElement> {
...
   return (
-    <Text as="label" htmlFor={htmlFor} color={fg}>
+    <Text as="label" htmlFor={htmlFor} color={fg} style={{ ...disabledStyles }}>
```
Comment (P0lip) [inline src/components/Label.tsx]:
```suggestion
    <Text as="label" htmlFor={htmlFor} color={fg} css={{ ...disabledStyles }}>
```

**Review by P0lip — APPROVED** [review]
Cool stuff!

File: src/components/Label.tsx (hunk 1 above)
Comment (billiegoose) [reply src/components/Label.tsx]: Will that work with the spread operator?
```js
{...null}
```
Whaaaat it works. Why does it work?

Comment (P0lip) [reply src/components/Label.tsx]: Didn't check the spec, but object rest spread operators works with any (or almost) any value. Boolean would work too, for sure.

File: src/components/Label.tsx (hunk 2 above)
Comment (billiegoose) [reply src/components/Label.tsx]: OK... so we need to use `css` to benefit from emotion. Gradually I will absorb this knowledge lol.

**Pull request merged**